lightning/src/ln/channel.rs

```diff
@@ -1189,9 +1189,9 @@ impl_writeable_tlv_based!(PendingChannelMonitorUpdate, {
 pub(super) enum ChannelPhase<SP: Deref> where SP::Target: SignerProvider {
 	UnfundedOutboundV1(OutboundV1Channel<SP>),
 	UnfundedInboundV1(InboundV1Channel<SP>),
-	#[cfg(dual_funding)]
+	#[cfg(any(dual_funding, splicing))]
 	UnfundedOutboundV2(OutboundV2Channel<SP>),
```

Comment:
I'd like to see how this plays with splicing directly. What do we need to extract from these types that splicing shares?

Reply:
In the current prototype the channel is cycled back to `UnfundedOutboundV2` during splicing -- until the new funding is confirmed the channel is like an unfunded channel. This may solution may not be final, admittedly.
(ref https://github.com/optout21/rust-lightning/blob/splicing-hapa8/lightning/src/ln/channelmanager.rs#L8459)